### ISSUE TITLE
Unable to filter dimension in chart by double click - after update from Supserset 2.0.1 to 2.1.0 

### ISSUE THREAD

**Issue by PrinceGallywix** [apache/superset#23735]
After update to 2.1.0 from 2.0.1 we are unable to filter one line on chart with multiple dimensions via double clicking on it. (Does not depends on chart type: for example, Time-series Area chart and time-series bar chart had been tested).

It was very useful feature and all of our charts were depended on it (on version 2.0.1) - our users would simply double click on a single line to explore certain tendency rather then disabling every other line via navigation bar on top-right of the chart.

Are there any options to turn this feature back on? 

#### How to reproduce the bug

1. Go to any chart (for example, 'Time-series Area chart') with multiple dimensions
2. Double click on a single row

### Expected results

Only chosen dimension left on a chart 

### Actual results

Nothing happens, there are still multiple dimensions on a chart.

### Environment
- browser type and version: Google Chrome 109.0.5414.120 (also tested on Firefox 111.0.1)
- superset version: superset 2.1.0
- python version: python 3.8.12
- node.js version: node 10.24.0

### Checklist

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by xupeng** [comment]
Same issue here.

**Comment by PrinceGallywix** [comment]
Well, I'm trying to find out the reason this feature isn't working.

I'm not a frontend developer and sorry for possible confusing, but here are some results:
Comparing 2.1.0 with 2.0.1 in debug mode will show that EchartInstance reference won't be got anymore, I guess the problem in the reference handler:

https://github.com/apache/superset/blob/357745f418d3eb42fececd6fd7d4a59337709a1d/superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx#L59

In 2.0.1 echartRef will be equal to ChartInstance object, but in 2.1.0 you will get _undefined_ 

@villebro I see some of your activities in this module. Do you have any ideas how to fix this bug? Could you share your thoughts, please?

As you can see, there are plenty of "DoubleClick feature" users :)

**Comment by rusackas** [comment]
The current era ECharts (line, area, or otherwise) don't have a double-click feature. That sounds like what existed on the "legacy" (deprecated) line charts based on NVD3. Those still exist on 2.1.0, but will be migrated in 3.0. 

What is NOT release in 2.1.0, but is available on `master` and will be available in 3.0 as well, are the ECharts equivalent, the All and Inverse buttons you see here.
<img width="744" alt="image" src="https://github.com/apache/superset/assets/812905/04bf4b6f-543c-4c8f-804a-502e94eb57f2">

ECharts does not and will not support double click actions like this (they're not discoverable to new users, arguably), so we're supporting the ECharts-native solution instead. 

Hope this explains the situation... I'll close this for now, but if I'm misinterpreting the issue, let me know and we can re-open.

**Comment by PrinceGallywix** [comment]
@rusackas thank you for reply! We are excited to try this  newfeature with 'All' and 'Invert' buttons!

But I still don't get it...
The double click feature is described in EchartsTimeseries module (the link in my previous comment), which one is NOT marked as deprecated and implemented in Echarts module. So, double click feature is in ECharts as well, isn't it?

**Comment by rusackas** [comment]
> @rusackas thank you for reply! We are excited to try this newfeature with 'All' and 'Invert' buttons!
> 
> But I still don't get it... The double click feature is described in EchartsTimeseries module (the link in my previous comment), which one is NOT marked as deprecated and implemented in Echarts module. So, double click feature is in ECharts as well, isn't it?
> 
> UPD: Here is the correct link to the function:
> 
> https://github.com/apache/superset/blob/357745f418d3eb42fececd6fd7d4a59337709a1d/superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx#L72

Yep, I see that in the code, and it looks like it ought to work, but I don't see it actually _doing_ anything. That probably ought to be debugged or removed, one way or the other.

**Comment by PrinceGallywix** [comment]
@rusackas yep, it ought to work and it was OK in 2.0.1, but not in 2.1.0:

I've tryed to debug this feature, but then I got stuck. You can debug this in the next steps:
1. Open any time-series Echart (for example, open 'Time-series Bar Chart v2') with at least 2 dimensions,
2. Set a breakpoint on this line:
https://github.com/apache/superset/blob/357745f418d3eb42fececd6fd7d4a59337709a1d/superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx#L75
(on the user side, the path is: _webpack://superset/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx_ )
3. Double click on any bar (or on the name of dimention in annotation layer) 
4. Ok, now you are paused on a breakpoint
5. Check out the _echartInstance_ variable (which one is defined in the line above our debugging point).  In 2.1.0 it is _undefined_. I guess this is the problem, the reference handler is broken somewhere, but I don't know where and how to fix it.

This is how it was looking like in 2.0.1:
![image](https://github.com/apache/superset/assets/77333963/307eb039-163a-4026-a534-5faf92a22dab)

In 2.1.0 we have _undefined_. This is because the _echartRef_ is null, which one is defined a little above.

If you have any ideas about this bug, please share :)

I guess we can re-open this issue, because it's not deprecated and it would be very helpful for us to use it until 3.0 release

**Comment by PrinceGallywix** [comment]
Just for the record: this is where the feature is described and implemented previously:
https://github.com/apache/superset/issues/16354

**Comment by PrinceGallywix** [comment]
@rusackas Can we re-open the issue, please?

**Comment by michael-s-molina** [comment]
This was fixed in https://github.com/apache/superset/pull/24477 and will be available in 3.0

**Comment by Krknv** [comment]
@PrinceGallywix you can try to add **ref={echartRef}** to fix issue.

https://github.com/apache/superset/blob/357745f418d3eb42fececd6fd7d4a59337709a1d/superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx#L294

```jsx
<Echart
  ref={echartRef}
  refs={refs}
  height={height - extraControlHeight}
  width={width}
  echartOptions={echartOptions}
  eventHandlers={eventHandlers}
  zrEventHandlers={zrEventHandlers}
  selectedValues={selectedValues}
/>
```
